packages/host/app/components/card-catalog/modal.gts

```diff
@@ -409,147 +431,139 @@ export default class CardCatalogModal extends Component<Signature> {
-    }
-    let maybeIndexCardURL = this.cardService.realmURLs.find(
-      (u) => u === cardURL + '/',
-    );
-    const cardResource = getCard(this, () => maybeIndexCardURL ?? cardURL, {
-      isLive: () => false,
-    });
-    await cardResource.loaded;
-    let { card } = cardResource;
-    if (!card) {
-      this.setErrorState(`Could not find card at ${this.state.searchKey}`);
-      return;
-    }
-    let realmInfo = await this.cardService.getRealmInfo(card);
-    if (!realmInfo) {
-      this.setErrorState(`Encountered error getting realm info for ${cardURL}`);
-      return;
-    }
-    this.state.searchResults = [
-      {
-        url: card.id,
-        realmInfo,
-        cards: [card],
-      },
-    ];
-  });
-
   @action
   onSearchFieldUpdated() {
     if (!this.state) {
       return;
     }
+
     this.state.errorMessage = '';
 
-    if (!this.state.searchKey && !this.state.selectedRealms.length) {
+    if (!this.state.searchKey) {
       return this.resetState();
     }
 
-    if (this.searchKeyIsURL) {
-      this.getCard.perform(this.state.searchKey);
+    let searchKeyIsURL = this.searchKeyIsURL;
+    if (searchKeyIsURL) {
+      // This is when a user has entered a URL directly into the search field
+      // 1. if .json is missing, add it
+      // 2. if the URL points to a realm, add /index.json for convenience of  getting the index card
+      this.state.query = this.state.originalQuery;
+      let cardUrlFromSearchKey = this.state.searchKey;
+
+      if (
+        this.state.availableRealmUrls.some(
+          (url) =>
+            url === cardUrlFromSearchKey || url === cardUrlFromSearchKey + '/',
+        )
+      ) {
+        cardUrlFromSearchKey = cardUrlFromSearchKey.endsWith('/')
+          ? cardUrlFromSearchKey + 'index'
+          : cardUrlFromSearchKey + '/index';
+      }
+      this.state.cardUrlFromSearchKey =
+        cardUrlFromSearchKey +
+        (cardUrlFromSearchKey.endsWith('.json') ? '' : '.json');
       return;
     }
 
-    let results: RealmCards[] = [];
-    let cardFilter = (c: CardDef) => {
-      return !this.state
-        ? false
-        : c.title
-            ?.trim()
-            .toLowerCase()
-            .includes(this.state.searchKey.trim().toLowerCase());
-    };
-
-    for (let { url, realmInfo, cards } of this.displayedRealms) {
-      let filteredCards = cards.filter(cardFilter);
-      if (filteredCards.length) {
-        results.push({
-          url,
-          realmInfo,
-          cards: filteredCards,
-        });
+    let newFilter: EveryFilter | undefined;
+
+    if (this.state.originalQuery.filter) {
+      let _isCardTypeFilter = isCardTypeFilter(this.state.originalQuery.filter);
+      let _isEveryFilter = isEveryFilter(this.state.originalQuery.filter);
+
+      if (_isCardTypeFilter) {
+        newFilter = {
+          on: (this.state.originalQuery.filter as CardTypeFilter).type,
+          every: [{ contains: { title: this.state.searchKey } }],
```

Comment:
Previously, filtering by card title was done in the client. Now we are moving this to the server which will be better for when we'll want to add pagination in the server query